src/pspm_con1.m

```diff
@@ -130,26 +130,27 @@ function pspm_con1(modelfile, connames, convec, datatype, deletecon, options)
   % there are issues if data.stats has NaN and the corresponding conmat
   % also contains 0
   idx_issue = any(isnan(data.stats),2);
-  l_conmat_r = length(conmat(:,1)); 
+  l_conmat_r = length(conmat(:,1));
+  data_stats_converted = data.stats;
+  conmat_converted = conmat;
   if any(idx_issue(:))
     for i_conmat_r = 1:l_conmat_r % if conmat is a 2D matrix, repeat for each row
       conmat_array = conmat(i_conmat_r,:);
       idx_valid_issue = transpose(conmat_array==0) .* idx_issue;
       idx_invalid_issue = transpose(conmat_array~=0) .* idx_issue;
-      if any(idx_valid_issue) && ~any(idx_invalid_issue)
-        warning(['Calculated data.stats contain NaNs that are caused by unknown reasons. '...
-          'data.stats are then used in a contrasts, producing an invalid results (NaN).']);
-        data_stats_converted = data.stats;
-        data_stats_converted(isnan(data_stats_converted)) = 0;
-        conval = conmat * data_stats_converted;
-      else
+      if any(idx_valid_issue)
         warning(['Calculated data.stats contain NaNs that are caused by unknown reasons. '...
           'However they were not used in the computation of the contrasts.']);
-        conval = conmat * data.stats;
+        data_stats_converted_array = data_stats_converted(idx_valid_issue==1,:);
+        data_stats_converted_array(isnan(data_stats_converted_array)) = 0;
+        data_stats_converted(idx_valid_issue==1,:) = data_stats_converted_array;
+      elseif any(idx_invalid_issue)
+        warning(['Calculated data.stats contain NaNs that are caused by unknown reasons. '...
+          'data.stats are then used in a contrasts, producing an invalid result (NaN).']);
       end
     end
   end
-
+  conval = conmat_converted * data_stats_converted;
```

Comment:
I don't think this will provide the correct response for multiple contrast.
It seems to me that data_stats_converted keeps getting rewritten at each contrast.

For instance, if the contrast are [ 0, 1] and [1, 0] and the data have NaN in the first line.
With the current code the NaN will be converted to zero by the processing of the first contrast, and the computation of the second contrast will then not produce the NaN it ought to produce.

Reply:
This is the old version and has been removed, but thanks for reminding me of it 🙂